graviton/invariant.py

```diff
@@ -10,14 +11,47 @@
 )
 
 
+class PredicateKind(Enum):
+    Constant = "constant"
+    CaseMap = "case map"
+    ExactMatch = "exact match"
+    RangeMatch = "range match"
+    IdenticalPair = "identical"
+
+
 INVARIANT_TYPE = Union[
     PyTypes,
     Dict[Sequence[PyTypes], PyTypes],
     Callable[[PyTypes], PyTypes],
     Callable[[PyTypes], bool],
+    PredicateKind,  # currently represents the equality of two dry-run executable functions
 ]
 
 
+def get_kind(predicate: INVARIANT_TYPE) -> PredicateKind:
```

Comment:
this function seems like a type checking or a reducer, that maps `INVARIANT_TYPE` to `PredicateKind`.

Reply:
This is originally some logic in `prepare_predicate`, got refactored out for better readability.